### PR TITLE
[docs] Update index and quicktour

### PR DESCRIPTION
**Pull request by stevhliu**
Starts implementing some of the things we discussed in our docs plan:

- clarify the difference between using `AutoPeftModel` (inference) and `PeftConfig` + `get_peft_model` (train) and splits the quicktour more generally along these lines
- add API reference for `AutoPeftModel` classes
- cleans up the index with a Space for discovering what models/methods are supported given a task (let me know if you think this is easier and cleaner)
- create separate subsection for LoRA guides

### CODE REVIEW

File: docs/source/_toctree.yml [hunk 1]
```diff
@@ -9,24 +9,26 @@
...
     title: P-tuning for sequence classification
-  - local: task_guides/dreambooth_lora
-    title: Dreambooth fine-tuning with LoRA
-  - local: task_guides/token-classification-lora
-    title: LoRA for token classification
-  - local: task_guides/int8-asr
-    title: int8 training for automatic speech recognition
-  - local: task_guides/semantic-similarity-lora
-    title: Semantic similarity with LoRA
+  - title: LoRA
```
Comment (BenjaminBossan) [inline docs/source/_toctree.yml:18]: I like this structure, nice.

Strangely, in the doc preview, this section shows as all capitalized "LORA" for me, not sure why. Not a big deal, just curious.

File: docs/source/index.md [hunk 2]
```diff
@@ -43,100 +41,9 @@ PEFT is seamlessly integrated with 🤗 Accelerate for large-scale models levera
...
-
-|   Model         | LoRA | Prefix Tuning  | P-Tuning | Prompt Tuning  | IA3 |
-| --------- | ---- | ---- | ---- | ----  | ----  |
-| SegFormer           | ✅  |   |   |   |   |
-
+<iframe
+	src="https://stevhliu-peft-methods.hf.space"
+	frameborder="0"
+	width="850"
+	height="450"
```
Comment (BenjaminBossan) [inline docs/source/index.md:48]: I have no idea about iframes, but can the height be dynamic, so that we don't have to scroll for longer lists?

**Review by BenjaminBossan — COMMENTED** [review]
Thanks, Steven, this looks like it will help give a better structure to the PEFT docs and make the intro easier to understand for new users.

I left  a couple of comments, please take a look.

File: docs/source/quicktour.md [hunk 3]
```diff
@@ -68,83 +62,103 @@ model.print_trainable_parameters()
...
+    print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0])
+
+"Preheat the oven to 350 degrees and place the cookie dough in the center of the oven. In a large bowl, combine the flour, baking powder, baking soda, salt, and cinnamon. In a separate bowl, combine the egg yolks, sugar, and vanilla."
+```
 
-```diff
-- from peft import PeftModel, PeftConfig, AutoPeftModel
-+ from peft import AutoPeftModel
-- from transformers import WhisperForConditionalGeneration
+For other tasks that aren't explicitly supported with an `AutoPeftModelFor` class - such as automatic speech recognition - you can still use the base [`AutoPeftModel`] class.
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/quicktour.md]: I think this can be something we put in the Tutorial section (this is good basic info for new users to learn since `AutoPeftModel` is just a wrapper around `PeftModel`)? For the quicktour though, I think we should aim for convenience to get users started as quickly as possible :)

File: docs/source/index.md [hunk 4]
```diff
@@ -43,100 +41,9 @@ PEFT is seamlessly integrated with 🤗 Accelerate for large-scale models levera
...
-
-As with image-to-text models, you should be able to apply LoRA to any of the [segmentation models](https://huggingface.co/models?pipeline_tag=image-segmentation&sort=downloads). 
-It's worth noting that we haven't tested this with every architecture yet. Therefore, if you come across any issues, kindly create an issue report.
-
-|   Model         | LoRA | Prefix Tuning  | P-Tuning | Prompt Tuning  | IA3 |
-| --------- | ---- | ---- | ---- | ----  | ----  |
-| SegFormer           | ✅  |   |   |   |   |
-
+<iframe
+	src="https://stevhliu-peft-methods.hf.space"
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/index.md:45]: Currently, the underlying [dataset](https://huggingface.co/datasets/stevhliu/supported-peft-models-methods) would need to be updated each time a new model is added or if a new method is supported. Let me know if you have any suggestions for automating it though, that'd be nice!

[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply docs/source/index.md:45]: Technically, it would be possible. For example, we could set up a cron job or GitHub CI to parse the README (assuming that should be the single source of truth) with pandoc, extract the table, and create an update the on the dataset. But that seems like overkill for now. Probably we, the maintainers, just need to remember to update the dataset each time a new model is supported.

**Review by BenjaminBossan — APPROVED** [review]
Thanks for improving the docs, LGTM. I think it would be good to have another approval by Sourab or Younes before merging.

File: docs/source/quicktour.md [hunk 5]
```diff
@@ -68,83 +62,108 @@ model.print_trainable_parameters()
...
-+ peft_model = AutoPeftModelForCausalLM.from_pretrained("ybelkada/opt-350m-lora")
-```
+model = model.to("cuda")
+model.eval()
+inputs = tokenizer("Preheat the oven to 350 degrees and place the cookie dough", return_tensors="pt")
 
-Currently, supported auto classes are: `AutoPeftModelForCausalLM`, `AutoPeftModelForSequenceClassification`, `AutoPeftModelForSeq2SeqLM`, `AutoPeftModelForTokenClassification`, `AutoPeftModelForQuestionAnswering` and `AutoPeftModelForFeatureExtraction`. For other tasks (e.g. Whisper, StableDiffusion), you can load the model with:
+with torch.no_grad():
+    outputs = model.generate(input_ids=inputs["input_ids"].to("cuda"), max_new_tokens=50)
+    print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0])
```
Comment (younesbelkada) [inline docs/source/quicktour.md:147]:
```suggestion
outputs = model.generate(input_ids=inputs["input_ids"].to("cuda"), max_new_tokens=50)
print(tokenizer.batch_decode(outputs.detach().cpu().numpy(), skip_special_tokens=True)[0])
```
The `torch.no_grad()` context manager is already used in `generate` AFAIK
